### ISSUE TITLE
Bug: Creating a multi-channel tiff image from scratch.

### ISSUE THREAD

**Issue by nishaq503** [PolusAI/bfio#35]
In the following snippet, we create a two-channel image from scratch and try to save it with the `BioWriter`. This raises an error abour the inability to set the `index` attribute of `tifffile.TiffFrame`.

```python
import pathlib

import bfio
import numpy

shape = (10240, 10240)
intensity = numpy.random.uniform(size=shape).astype(numpy.float32)
mask = (intensity > 0.5).astype(numpy.float32)
path = pathlib.Path(__file__).parent.joinpath('test.ome.tif')

with bfio.BioWriter(path) as writer:
    writer.dtype = numpy.float32
    writer.X = shape[0]
    writer.Y = shape[1]
    writer.ps_x = (1.5, 'mm')
    writer.ps_y = (1.5, 'mm')
    writer.C = 2
    writer.channel_names = ['intensity', 'mask']

    # axis ordering: (X ,Y, Z, C, T)
    writer[:, :, 0, 0, 0] = intensity[:]
    writer[:, :, 0, 1, 0] = mask[:]

with bfio.BioReader(path) as reader:
    image = reader[:]

print(image.shape, image.dtype)
```

Here's the stacktrace:

```bash
Traceback (most recent call last):
  File "./test_bfio.py", line 34, in <module>
    test(shape=(10240, 10240))
  File "./test_bfio.py", line 26, in test
    image = reader[:]
  File "./.venv/lib/python3.9/site-packages/bfio/bfio.py", line 212, in __getitem__
    return self.read(**ind)
  File "./.venv/lib/python3.9/site-packages/bfio/bfio.py", line 306, in read
    self._backend.read_image(
  File "./.venv/lib/python3.9/site-packages/bfio/base_classes.py", line 809, in read_image
    self._read_image(*args)
  File "./.venv/lib/python3.9/site-packages/bfio/backends.py", line 615, in _read_image
    offsets, bytecounts = self._chunk_indices(X, Y, Z, C, T)
  File "./.venv/lib/python3.9/site-packages/bfio/backends.py", line 556, in _chunk_indices
    dataoffsets, databytecounts = self._page_offsets_bytes(index)
  File "./.venv/lib/python3.9/site-packages/bfio/backends.py", line 530, in _page_offsets_bytes
    obc = self._TiffBytesOffsets(parent, index)
  File "./.venv/lib/python3.9/site-packages/bfio/backends.py", line 463, in __init__
    self.index = index
AttributeError: can't set attribute
```

The attribute in question is actually called `_index` instead of `index`.
So [line 463](https://github.com/PolusAI/bfio/blob/master/bfio/backends.py#L463) should probably be:
```python
self._index = [index]
```

After this fix, the next error about calling the method `tifffile.TiffTag.fromfile`.
Here's the stacktrace:
```bash
  File "./test_bfio.py", line 34, in <module>
    test(shape=(10240, 10240))
  File "./test_bfio.py", line 26, in test
    image = reader[:]
  File "./.venv/lib/python3.9/site-packages/bfio/bfio.py", line 212, in __getitem__
    return self.read(**ind)
  File "./.venv/lib/python3.9/site-packages/bfio/bfio.py", line 306, in read
    self._backend.read_image(
  File "./.venv/lib/python3.9/site-packages/bfio/base_classes.py", line 809, in read_image
    self._read_image(*args)
  File "./.venv/lib/python3.9/site-packages/bfio/backends.py", line 616, in _read_image
    offsets, bytecounts = self._chunk_indices(X, Y, Z, C, T)
  File "./.venv/lib/python3.9/site-packages/bfio/backends.py", line 557, in _chunk_indices
    dataoffsets, databytecounts = self._page_offsets_bytes(index)
  File "./.venv/lib/python3.9/site-packages/bfio/backends.py", line 531, in _page_offsets_bytes
    obc = self._TiffBytesOffsets(parent, index)
  File "./.venv/lib/python3.9/site-packages/bfio/backends.py", line 472, in __init__
    for code, tag in self._gettags({324, 325}):
  File "./.venv/lib/python3.9/site-packages/bfio/backends.py", line 506, in _gettags
    tag = tifffile.TiffTag.fromfile(
TypeError: fromfile() takes 2 positional arguments but 4 were given
```

This method requires that certain arguments be passed in by name.
We need to change [lines 508 and 509](https://github.com/PolusAI/bfio/blob/75c588d60cec3cdfc0a24886acd90c89a0703be6/bfio/backends.py#L506-L510):
```python
tag = tifffile.TiffTag.fromfile(
    self.parent,
    offset=tagoffset + tagindex,
    header=tagbytes[tagindex : tagindex + tagsize],
)
```

**Comment by JacksonMaxfield** [comment]
@Nicholas-Schaub I think this is happening due to the recent tifffile release. The recent release was a lot of large breaking changes and certain properties can't be set anymore. We are running into issues on CI with trying to upgrade now: https://github.com/AllenCellModeling/aicsimageio/runs/6178457018?check_suite_focus=true

**Comment by JacksonMaxfield** [comment]
@nishaq503 you may want to try installing an older version of tifffile

**Comment by Nicholas-Schaub** [comment]
@JacksonMaxfield Would you have a problem if we just restricted the tifffile version? Over the two or so years of bfio development, we have found that tifffile has frequently introduced changes that break bfio. I don't generally work on bfio weekly, so what I did was restrict the tifffile version until I have time to make the necessary updates every few months.

**Comment by JacksonMaxfield** [comment]
I think it's fine for you to pin. I just need to come up with better testing infra for aicsimageio to create separate installs / envs for each reader i think.

**Comment by Nicholas-Schaub** [comment]
Resolved in new development version #41
